File: auth0/src/main/java/com/auth0/android/provider/RedirectActivity.java
package com.auth0.android.provider;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import androidx.annotation.Nullable;

@SuppressLint("GoogleAppIndexingApiWarning")
public class RedirectActivity extends Activity {

    @Override
    public void onCreate(@Nullable Bundle savedInstanceBundle) {
        super.onCreate(savedInstanceBundle);
        Intent intent = new Intent(this, AuthenticationActivity.class);
        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_SINGLE_TOP);
        if (getIntent() != null) {
            intent.setData(getIntent().getData());
        }
        startActivity(intent);
        finish();
    }

}

File: auth0/src/main/java/com/auth0/android/provider/RedirectActivity.kt
package com.auth0.android.provider

import android.app.Activity
import android.content.Intent
import android.os.Bundle

public class RedirectActivity : Activity() {
    public override fun onCreate(savedInstanceBundle: Bundle?) {
        super.onCreate(savedInstanceBundle)
        val intent = Intent(this, AuthenticationActivity::class.java)
        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_SINGLE_TOP)
        if (getIntent() != null) {
            intent.data = getIntent().data
        }
        startActivity(intent)
        finish()
    }
}